syntax = "proto3";

package appointment;

option go_package = "github.com/somnathbm/hospital-hms/microservices/appointment-service/gen;gen";

service AppointmentService {
  rpc CheckAppointment (CheckAppointmentRequest) returns (CheckAppointmentResponse);
  rpc BookAppointment (BookAppointmentRequest) returns (BookAppointmentResponse);
}

// CheckAppointment
message CheckAppointmentRequest {
  string patient_id = 1;
  string appointment_id = 2;
}

message CheckAppointmentResponse {
  bool valid = 1;
  string msg = 2;
}

// BookAppointment
message BookAppointmentRequest {
  string patient_id = 1;
  string doctor_id = 2;
  string slot = 3;
}

message BookAppointmentResponse {
  string appointment_id = 1;
  string msg = 2;
}